File: Assets/ProjectGame/Lib/data/class_data_id/RoleType/RoleTypeTableID.js
// RoleTypeTableID - Pure JavaScript Version
export const RoleTypeTableID = {
    None: 0,  // デフォルト値（C#互換）
    Villager: 1,  // 村人
    Werewolf: 2,  // 人狼
    RoleType_02: 3,  // 
    RoleType_03: 4,  // 
    RoleType_04: 5,  // 
    RoleType_05: 6,  // 
    RoleType_06: 7,  // 
    RoleType_07: 8,  // 
    RoleType_08: 9,  // 
    RoleType_09: 10,  // 
    RoleType_10: 11,  // 
    Max: 12
};

export const RoleTypeTableIDExtensions = {
    /**
     * Enumを数値に変換
     */
    toInt(id) {
        return Number(id);
    },

    /**
     * 数値をEnumに変換
     */
    toRoleTypeTableID(id) {
        return id;
    },

    /**
     * 0-based indexに変換
     */
    toIndex(id) {
        return Number(id) - 1;
    },

    /**
     * すべてのIDに対して処理を実行
     */
    forID(action) {
        if (typeof action !== 'function') {
            throw new Error('action must be a function');
        }
        const start = RoleTypeTableID.Villager;
        const max = RoleTypeTableID.Max;
        for (let id = start; id < max; id++) {
            if (Object.values(RoleTypeTableID).includes(id)) {
                action(id);
            }
        }
    },

    /**
     * 条件に合うすべてのIDを返す
     */
    findAll(predicate) {
        if (typeof predicate !== 'function') {
            throw new Error('predicate must be a function');
        }
        const results = [];
        const start = RoleTypeTableID.Villager;
        const max = RoleTypeTableID.Max;
        for (let id = start; id < max; id++) {
            if (Object.values(RoleTypeTableID).includes(id)) {
                if (predicate(id)) {
                    results.push(id);
                }
            }
        }
        return results;
    },

    /**
     * 条件に合う最初のIDを返す（見つからなければ None）
     */
    find(predicate) {
        if (typeof predicate !== 'function') {
            throw new Error('predicate must be a function');
        }
        const start = RoleTypeTableID.Villager;
        const max = RoleTypeTableID.Max;
        for (let id = start; id < max; id++) {
            if (Object.values(RoleTypeTableID).includes(id)) {
                if (predicate(id)) {
                    return id;
                }
            }
        }
        return RoleTypeTableID.None;
    }
};

File: Assets/ProjectGame/Lib/data/class_data_id/RoleType/RoleTypeTableID.py
from enum import IntEnum
from typing import Callable, List, Any


class RoleTypeTableID(IntEnum):
    NONE = 0  # デフォルト値（C#のNoneに相当）
    Villager = 1  # 村人
    Werewolf = 2  # 人狼
    RoleType_02 = 3  # 
    RoleType_03 = 4  # 
    RoleType_04 = 5  # 
    RoleType_05 = 6  # 
    RoleType_06 = 7  # 
    RoleType_07 = 8  # 
    RoleType_08 = 9  # 
    RoleType_09 = 10  # 
    RoleType_10 = 11  # 
    MAX = 12

def to_int(id: "RoleTypeTableID") -> int:
    return int(id)


def to_RoleTypeTableID(id: int) -> "RoleTypeTableID":
    return RoleTypeTableID(id)


def to_index(id: "RoleTypeTableID") -> int:
    return int(id) - 1


def for_id(action: Callable[["RoleTypeTableID"], None]):
    if action is None:
        raise ValueError("action cannot be None")
    start = RoleTypeTableID.Villager.value
    for i in range(start, RoleTypeTableID.MAX.value):
        try:
            value = RoleTypeTableID(i)
            action(value)
        except ValueError:
            continue  # 未定義の値はスキップ


def find_all(predicate: Callable[["RoleTypeTableID"], bool]) -> List["RoleTypeTableID"]:
    if predicate is None:
        raise ValueError("predicate cannot be None")
    results: List["RoleTypeTableID"] = []
    start = RoleTypeTableID.Villager.value
    for i in range(start, RoleTypeTableID.MAX.value):
        try:
            value = RoleTypeTableID(i)
            if predicate(value):
                results.append(value)
        except ValueError:
            continue
    return results


def find(predicate: Callable[["RoleTypeTableID"], bool]) -> "RoleTypeTableID":
    if predicate is None:
        raise ValueError("predicate cannot be None")
    start = RoleTypeTableID.Villager.value
    for i in range(start, RoleTypeTableID.MAX.value):
        try:
            value = RoleTypeTableID(i)
            if predicate(value):
                return value
        except ValueError:
            continue
    return RoleTypeTableID.NONE
